tests/next_tests/unit_tests/iterator_tests/transforms_tests/test_global_tmps.py

```diff
@@ -6,464 +6,219 @@
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
 
-# TODO(tehrengruber): add integration tests for temporaries starting from manually written
-#  itir. Currently we only test temporaries from frontend code which makes testing changes
-#  to anything related to temporaries tedious.
-import copy
+from typing import Optional
 
-import gt4py.next as gtx
-from gt4py.eve.utils import UIDs
 from gt4py.next import common
-from gt4py.next.iterator import ir
+from gt4py.next.iterator import ir as itir
 from gt4py.next.iterator.ir_utils import ir_makers as im
-from gt4py.next.iterator.transforms.global_tmps import (
-    AUTO_DOMAIN,
-    FencilWithTemporaries,
-    SimpleTemporaryExtractionHeuristics,
-    collect_tmps_info,
-    split_closures,
-    update_domains,
-)
+from gt4py.next.iterator.transforms import global_tmps, infer_domain
+from gt4py.next.iterator.type_system import inference as type_inference
 from gt4py.next.type_system import type_specifications as ts
 
 
 IDim = common.Dimension(value="IDim")
 JDim = common.Dimension(value="JDim")
 KDim = common.Dimension(value="KDim", kind=common.DimensionKind.VERTICAL)
-index_type = ts.ScalarType(kind=getattr(ts.ScalarKind, ir.INTEGER_INDEX_BUILTIN.upper()))
+index_type = ts.ScalarType(kind=getattr(ts.ScalarKind, itir.INTEGER_INDEX_BUILTIN.upper()))
 float_type = ts.ScalarType(kind=ts.ScalarKind.FLOAT64)
 i_field_type = ts.FieldType(dims=[IDim], dtype=float_type)
 index_field_type_factory = lambda dim: ts.FieldType(dims=[dim], dtype=index_type)
 
 
-def test_split_closures():
-    UIDs.reset_sequence()
-    testee = ir.FencilDefinition(
-        id="f",
+def program_factory(
```

Comment:
Is this really the only placed where this factory is/should be used? It looks like a pretty general utility to me....